File: src/main/resources/static/js/contrato/contrato.js
/**
 * Create a CSV file from the results being showed.
 * @returns csv file.
 */
function createCSVFile(){
    var html = document.querySelector("#contratoTable").outerHTML;
    exportTableToCsv(html, "contrato.csv");
}

/**
 * 
 * @returns
 */
function createDocFile(){
	
	var content = document.querySelector('.docFile').outerHTML;
	var converted = htmlDocx.asBlob(content, {orientation: 'landscape'});
	saveAs(converted, 'contrato.docx');
}

/**
 * Create a PDF file from the filtered table.
 * @param licitacao table.
 * @returns
 */
function createPDFFile(contrato){
	
	var columns = ["Numero", "Fornecedor", "Orgão", "Data Assinatura", "Data Vigência", "Data Publicação", "Valor"];
	var pdfData = [];
	
	contrato.forEach(function (value, i){
		pdfData.push( [value.numero, value.fornecedor.razaoSocial, value.orgao.orgao, value.dataAssinatura, value.dataVigencia, value.dataPublicacao, value.valor]);
	});
	
	var doc = new jsPDF('l');
    doc.setFontSize(8);
    doc.autoTable(columns, pdfData, {
    	 
    	styles: {overflow: 'linebreak'},
    	columnStyles: {
    		0: {columnWidth: 25},
    		1: {columnWidth: 60},
    		2: {columnWidth: 60},
    		3: {columnWidth: 22},
    		4: {columnWidth: 22},
    		5: {columnWidth: 22},
    		6: {columnWidth: 25}
    	}
    });
    doc.output("dataurlnewwindow");
}

File: src/main/java/br/com/prefeitura/web/model/Contrato.java
package br.com.prefeitura.web.model;

import lombok.Builder;
import lombok.Data;

@Data
public class Contrato {

	private Long id;
	private String numero;
	private String dataAssinatura;
	private String dataVigencia;
	private String dataPublicacao;
	private String valor;
	private String objeto;
	private Fornecedor fornecedor;
	private Orgao orgao;
	
	
	private String dataInicialVO;
	private String dataFinalVO;
	private Long orgaoVO;
	private String cnpjVO;
	private String fornecedorVO;
	
	
	public Contrato(Long id, String numero, String dataAssinatura, String dataVigencia, String dataPublicacao,
			String valor, String objeto, Fornecedor fornecedor, Orgao orgao) {
		this.id = id;
		this.numero = numero;
		this.dataAssinatura = dataAssinatura;
		this.dataVigencia = dataVigencia;
		this.dataPublicacao = dataPublicacao;
		this.valor = valor;
		this.objeto = objeto;
		this.fornecedor = fornecedor;
		this.orgao = orgao;
	}
	
	public Contrato() {
		super();
	}

	/**
	 * 
	 * @param id
	 * @param numero
	 * @param dataAssinatura
	 * @param dataVigencia
	 * @param dataPublicacao
	 * @param valor
	 * @param objeto
	 * @param fornecedor
	 * @param orgao
	 * @return
	 */
	@Builder(builderMethodName = "contratoBuilder")
	public static Contrato newContrato(Long id, String numero, String dataAssinatura, String dataVigencia, String dataPublicacao,
			String valor, String objeto, Fornecedor fornecedor, Orgao orgao) {
		Contrato contrato = new Contrato(id, numero, dataAssinatura, dataVigencia, dataPublicacao, valor, objeto, fornecedor, orgao);
		return contrato;
	}
}
